gbpservice/nfp/configurator/drivers/firewall/vyos/vyos_fw_driver.py

```diff
@@ -565,43 +500,28 @@ def create_firewall(self, context, firewall, host):
                " %r. URL: %s" % (firewall['id'], firewall['tenant_id'], url))
         LOG.info(msg)
         data = jsonutils.dumps(firewall)
-        try:
-            resp = requests.post(url, data, timeout=self.timeout)
-        except requests.exceptions.ConnectionError as err:
-            self._print_exception('ConnectionError', err, url, 'create')
-            raise requests.exceptions.ConnectionError(err)
-        except requests.exceptions.RequestException as err:
-            self._print_exception('RequestException', err, url, 'create')
-            raise requests.exceptions.RequestException(err)
 
-        msg = ("POSTed the configuration to Service VM")
-        LOG.debug(msg)
-        if resp.status_code in common_const.SUCCESS_CODES:
-            try:
-                resp_payload = resp.json()
-                if resp_payload['config_success']:
-                    msg = ("Configured Firewall successfully. URL: %s"
-                           % url)
-                    LOG.info(msg)
-                    return common_const.STATUS_ACTIVE
-                else:
-                    self._print_exception('Failure',
-                                          resp.status_code, url,
-                                          'create', resp.content)
-                    return common_const.STATUS_ERROR
-            except ValueError as err:
-                self._print_exception('ValueError', err, url,
-                                      'create', resp.content)
-                return common_const.STATUS_ERROR
-            except Exception as err:
-                self._print_exception('UnexpectedError', err, url,
-                                      'create', resp.content)
-                return common_const.STATUS_ERROR
-        else:
-            self._print_exception('Failure', resp.status_code, url,
-                                  'create', resp.content)
+        err_msg = ("Configure firewall POST request to the VyOS "
+                   "service at %s failed. " % url)
+        try:
+            resp = self.rest_api.fire(url, data, common_const.POST)
+        except Exception as err:
+            err_msg += ("Reason: %r" % str(err).capitalize())
+            LOG.error(err_msg)
             return common_const.STATUS_ERROR
 
+        if resp is common_const.STATUS_SUCCESS:
+            msg = ("Configured firewall successfully for service at %r." % url)
+            LOG.info(msg)
+            return common_const.STATUS_ACTIVE
+
+        err_msg += (("Reason: %r, Response Content: %r" %
+                     (resp.pop('message'), resp))
```

Comment:
Resp is constructed in the RestApi class. It can either be a string or a dict. I don't think that this will be an issue. Checked it.